ui/src/BaseEditor.ts

```diff
@@ -0,0 +1,166 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { UserCancelledError } from './errors';
+import { localize } from "./localize";
+import { TemporaryFile } from './utils/TemporaryFile';
+
+// tslint:disable-next-line:typedef
+const dialogResponses = {
+    OK: localize('OK', "OK"),
+    DontShowAgain: localize('dontShow', "Don't Show Again")
+};
+
+// tslint:disable-next-line:no-unsafe-any
+export abstract class BaseEditor<ContextT> implements vscode.Disposable {
+    private fileMap: { [key: string]: [vscode.TextDocument, ContextT] } = {};
+    private ignoreSave: boolean = false;
+
+    constructor(readonly showSavePromptKey: string, readonly outputChanel?: vscode.OutputChannel) {
+    }
+
+    public abstract getData(context: ContextT): Promise<string>;
+    public abstract updateData(context: ContextT, data: string): Promise<string>;
+    public abstract getFilename(context: ContextT): Promise<string>;
+    public abstract getSize(context: ContextT): Promise<number>;
+    public abstract getSaveConfirmationText(context: ContextT): Promise<string>;
+
+    public async showEditor(context: ContextT): Promise<void> {
+        const size: number = await this.getSize(context);
+        const fileName: string = await this.getFilename(context);
+        const splitFileName: string[] = fileName.split('.');
+        const extension: string = splitFileName[splitFileName.length - 1];
+
+        this.appendToOutput(localize('opening', 'Opening "{0}" ...', fileName));
+        if (size > 50 /*Megabytes*/) {

```

Comment:
Why do we set a hard-coded limit of 50? Shouldn't Azure return a readable error if it's too large? Maybe different pricing tiers have different limits anyways

You should at least let the baseEditor overwrite the limit (I'm guessing cosmos has different limits than storage)

Reply:
Well, I think this is more of a vscode.Editor limitation rather than an App Service Plan tier limitation.  From what I understand, @JakeRadMSFT tested the size limit of what the editor could open without crashing.

Reply:
I've attached what the logic is for a file that is too large.  I'm okay with removing these lines; I'm not sure that this is a great error message though.

![image](https://user-images.githubusercontent.com/5290572/33863211-854c118c-de9b-11e7-8a2b-3839acca0878.png)

Reply:
Okay-- talked to Jake.  50 MB is a limit due reduce wait time.  I'll allow an optional parameter to check for this.